client/tests/unit/CosmologyForm.spec.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-await-in-loop, no-restricted-syntax, no-continue */
+
+import { mount, createLocalVue } from '@vue/test-utils';
+import CosmologyForm from '@/components/forms/CosmologyForm';
+import BACKEND_CONSTANTS from '@/constants/backend_constants';
+import INITIAL_STATE from '@/constants/initial_state';
+
+describe('Mounted CosmologyForm', () => {
+  const localVue = createLocalVue();
+  const wrapper = mount(CosmologyForm, {
+    propsData: {
+      title: 'Cosmology',
+      init: INITIAL_STATE.cosmology,
+    },
+    localVue,
+  });
+
+  const options = Object.keys(BACKEND_CONSTANTS.Cosmo_params);
+
+  test('has correct default model', async () => {
+    await wrapper.vm.$nextTick();
+    expect(wrapper.vm.model.cosmo_model).toBe('Planck15');
+  });
+
+  test('changes model values when values are changed', async () => {
+    const model = wrapper.vm.$data.model;
+    Object.keys(model).forEach(async (key) => {
+      if (typeof model[key] === 'number') {
+        const o = wrapper.vm.model[key];
+        model[key] += 1;
+        await localVue.nextTick();
+        await localVue.nextTick();
+        const n = wrapper.vm.model[key];
+        expect(n).toBe(o+1);
+      }
+    })
+  });
```

Comment:
This isn't testing anything meaningful. You're modifying an object and then verifying that the object was modified.

Reply:
A couple lines down there's the exact same test that tests params. They're not the same object. Line 26 is $data.model Line 33 is model itself, with things changing in data.
This tests if vmodel does anything

There's another test that tests if things get emitted when vmodel changes a couple functions below.

Reply:
wrapper.vm.$data.model == wrapper.vm.model. They're the same object (reference and all). vue-test-utils gives you getters and setters at the vm level so you don't have to go through $data to get to the object. 

The test you have modifies one of the model objects keys and sees if that 
The test that follows changes cosmo_model and sees if the sub parameters have updated.
The test I'm asking for changes cosmo_model and sees if the core parameters have update.